### PR TITLE
Improve flexibility for computation of statistics timeseries in AeroVal

### PR DESCRIPTION
**Pull request by jgliss**
This PR aims at fixing #409.

It will enable to specify the data frequency to be used as input for calculation of the statistics timeseries in a certain output resolution. E.g. for now input resolution is the same as output resolution (default monthly, can be specified via `pyaerocom.aeroval.setupclasses.TimeSetup.main_freq`), so for any given month, statistics were computed based on 1value for any given station (thus, 100 datapoints for each month for a dataset containing 100 stations). This will remain the default behaviour, however, this PR will add a new analysis option `stats_tseries_base_freq` in `pyaerocom.aeroval.setupclasses.StatisticsSetup`, which can be used to use a higher resolution as a basis for computation of the statistics in the desired output resolution (e.g. `daily` as required for #409).

NOTE: to be merged after PR #414.

### CODE REVIEW

**Comment by jgliss** [comment]
@avaldebe okay, if I add a test module aeroval/test/test_coldatatojson_engine? Want to start adding a few tests for parts of the code in that module.

Or did you already add such a module in one of your branches?

File: pyaerocom/aeroval/setupclasses.py [hunk 1]
```diff
@@ -70,10 +70,55 @@ def __init__(self, **kwargs):
...
+        days, obviously).
+
+
+    Parameters
+    ----------
+    kwargs
+        any of the supported attributes.
+
+    """
     MIN_NUM = 3
```
Comment (jgriesfeller) [inline pyaerocom/aeroval/setupclasses.py:117]: minor comment: I would prefer to see a small comment of the usage of this variable. Not a reason to not merge!

Comment (jgriesfeller) [reply pyaerocom/aeroval/setupclasses.py:117]: ready to be merged

Comment (jgliss) [reply pyaerocom/aeroval/setupclasses.py:118]: Done!

**Comment by jgliss** [comment]
@jgriesfeller I added some tests for the statistics timeseries computation, and I think this PR is now good to go. Could you have a quick look at the latest changes please? Cheers

**Review by jgriesfeller — APPROVED** [review]
all looks fine.

**Comment by jgliss** [comment]
@jgriesfeller I added some tests for the statistics timeseries computation, and I think this PR is now good to go. Could you have a quick look at the latest changes please? Cheers

> all looks fine.

Thanks!

**Pull request merged**